scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,399 @@
+        // Set completion time based on pbereports.log file
+        dto.setCompletionDate(latestCompletionTime.get());
+      } else {
+        // We don't have the pbereport.log, we'll have to fallback on using .transferdone file
+        // creation time instead
+        if (dto.getCompletionDate() == null) {
+          setCompletionTimeFromTransferDone(runDirectory, dto);
+        }
+      }
+    } else {
+      // There are some missing files, the run may not be complete
+      dto.setHealthType(HealthType.RUNNING);
+      log.warn("There may be some missing .transferdone and Transfer_Test files");
+    }
+
+    return dto;
+  }
+
+  /**
+   * Parse a metadata XML file and put all the relevant data into the DTO.
+   *
+   * @param metadata the path to the XML file
+   * @param dto the DTO to update
+   */
+  private void processMetadata(Document metadata, PacBioNotificationDto dto, TimeZone timeZone) {
+    for (RevioPacBioProcessor.ProcessMetadata processor : REVIO_METADATA_PROCESSORS) {
+      try {
+        processor.accept(metadata, dto, timeZone);
+      } catch (XPathException e) {
+        log.error("Failed to extract metadata", e);
+      }
+    }
+  }
+
+  /**
+   * Tests whether a String is blank (empty or just spaces) or null. Duplicated from MISO's
+   * LimsUtils.
+   *
+   * @param s String to test for blank or null
+   * @return true if blank or null String provided
+   */
+  private static boolean isStringBlankOrNull(String s) {
+    return s == null || s.trim().isEmpty();
+  }
+
+  @Override
+  public PathType getPathType() {
+    return PathType.DIRECTORY;
+  }
+
+  /**
+   * Returns a stream that includes Metadata directory level to be used for further processing
+   *
+   * @param runDirectory which we are currently processing
+   * @param directoryName specific sub-directory we want
+   * @return Stream from Metadata directory level
+   */
+  private Stream<File> streamSmrtCellSubdirectories(File runDirectory, String directoryName) {
+    return Stream.of(runDirectory.listFiles())
+        .filter(
+            cellDirectory ->
+                cellDirectory.isDirectory() && REVIO_CELL_DIRECTORY.test(cellDirectory.getName()))
+        .flatMap(
+            cellDirectory ->
+                Stream.of(cellDirectory.listFiles())
+                    .filter(
+                        subDirectory ->
+                            subDirectory.isDirectory()
+                                && subDirectory.getName().equals(directoryName)));
+  }
+
+  /**
+   * Returns the time a file was created
+   *
+   * @param file which we are currently processing
+   * @return Instant point in time
+   */
+  private static Instant getFileCreationTime(File file) {
+    try {
+      FileTime fileTime =
+          (FileTime) Files.getAttribute(Path.of(file.getAbsolutePath()), "creationTime");
+      return fileTime.toInstant();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
Change this method to throw `IOException` instead of catching it. The `process` method is already expected to throw it, so it's fine to let this bubble up

Reply:
Missed this one

Reply:
The easiest good solution would be to do this outside of a stream